examples/platform/telink/common/src/AppTaskCommon.cpp

```diff
@@ -108,6 +109,22 @@ Identify sIdentify = {
 uint8_t sTestEventTriggerEnableKey[TestEventTriggerDelegate::kEnableKeyLength] = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
                                                                                    0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff };
 #endif
+
+class AppCallbacks : public AppDelegate
+{
+    bool isComissioningStarted;
+
+public:
+    void OnCommissioningSessionStarted() override { isComissioningStarted = true; }
+    void OnCommissioningSessionStopped() override { isComissioningStarted = false; }
+    void OnCommissioningWindowClosed() override
+    {
+        if (!isComissioningStarted)
```

Comment:
How was this tested?  What scenarios were tested?

Reply:
@bzbarsky-apple 
This was tested with the following scenarios:
1. Start commissioning over BLE and pass this step. (Success)
2. Start commissioning over BLE and break it. The MCU should return to advertising state (Success)
3. Wait 15 min till commissioning window closes and check if the BLE is disabled (Success)

Our goal is to switch off the BLE only in case if commissioning window is closed by timeout. So no pairing tries from user in 15 minutes.
Maybe you have any ideas how to get the reason, why the commissioning window was closed?

Reply:
If you start commissioning over BLE and then commissioning completes successfully, do you want to switch off BLE?

> Maybe you have any ideas how to get the reason, why the commissioning window was closed?

Well, reviving https://github.com/project-chip/connectedhomeip/pull/25215 or something like that might work....

Reply:
@bzbarsky-apple 
If the commissioning windows has timed out, I want to switch off the BLE completely.
If you remember, the CHIP project has the 15 minutes timeout in case of inactivity after first startup (I mean if the board is completely new, or had the factory reset, and the user have not tried to perform any kind of commissioning).
After this 15 min timer expires, the Commissioning window closes. 
I need to handle only this one case and switch off the BLE only in this case. Because in Telink MCU switching off the advertisement and if there is no active session doesn't mean that the BLE is inactive. It is still consuming the CPU time and energy. We need to reduce the energy consumption in this case.

Reply:
@mishadesh OK, so you don't care about turning off BLE if commissioning in fact completes successfully (after which BLE is likely not used again)?  Or does something else handle turning off BLE in that case?

Reply:
@bzbarsky-apple 

I case if commissioning passed successfully, the Thread credentials were received and stored in memory.
The BLE manager checks if credentials are available on _HandleBleConnectionClosed_ event.
If they are available, the RF module switches off the BLE and switches on Thread.

![ble_conn_close](https://github.com/project-chip/connectedhomeip/assets/36621698/7e094361-33fa-481f-956f-888e3c085a70)

Reply:
OK.  I think it's worth adding comments here that explain:

1.  Why you only care about the case when commissioning fails and can't be restarted.
2. Why the logic implemented here detects exactly that case.

Because this seems to be depending on un-documented implementation details of thecommissioning window manager (e.g. that it will dispatch OnCommissioningWindowClosed when the fail-safe times out even if it has already dispatched on PASE establishment, etc).... It would be good to have those dependencies clear.

And a followup issue for having an API surface that does not require such dependencies would be good.

1.  Commissioning window is opened for 15 minute duration.
2. 14 minutes and 30 seconds pass.
3. A PASE session is set up over the next 20 seconds.  As part of OnSessionEstablished this will call OnCommissioningWindowClose

Reply:
@bzbarsky-apple 

https://github.com/project-chip/connectedhomeip/issues/27002

I have added the new feature request, please merge this pull request.